src/virtual-machine.js

```diff
@@ -226,26 +228,40 @@ class VirtualMachine extends EventEmitter {
 
         // Validate & parse
         if (typeof json !== 'string' && typeof json !== 'object') {
-            log.error('Failed to parse project. Invalid type supplied to fromJSON.');
-            return;
+            const invalidTypeError = 'Failed to parse project. Invalid type supplied to fromJSON.';
+            log.error(invalidTypeError);
```

Comment:
If you're throwing it, then it shouldn't log at this point, since the thrown error may eventually be caught and dealt with (possibly by logging the error).